iceoryx2-bb/posix/src/process_state.rs

```diff
@@ -0,0 +1,808 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Process monitoring via holding a file lock of a specific file. If the process crashes the
+//! lock will be released by the operating system and another process can detect the crash. If the
+//! process shutdowns correctly the file is removed and another process detects the clean shutdown.
+//!
+//! # Example
+//!
+//! ## Monitored Process
+//!
+//! ```
+//! use iceoryx2_bb_posix::process_state::*;
+//!
+//! let process_state_path = FilePath::new(b"process_state_file").unwrap();
+//!
+//! // remove leftovers from a previous abnormal process termination
+//! //
+//! // # Safety
+//! //  * This process owns the state file, therefore it is safe to remove a leftover from a
+//! //    previous run.
+//! unsafe { ProcessGuard::remove(&process_state_path).expect("") };
+//!
+//! // start monitoring from this point on
+//! let guard = ProcessGuard::new(&process_state_path).expect("");
```

Comment:
This I explicitly do not want. When the file still exists, it means someone must take care of the stale resources first before calling remove.

If one would have the ability to ignore that and just remove the file, they would violate the contract.

Removing this file is only allowed when the whole self-healing system crashed - with all of its processes. Then we need to clean up everything. Otherwise, one should never call remove.

I will adjust the example and make this more clear.

Reply:
You are right. I totally overlooked that. This is just a building block for the actual functionality